### PR TITLE
Further expand coverage of addmm/addmv, fix 0 stride

### PR DESCRIPTION
**Pull request by zasdfgbnm**
- test beta=0, self=nan
- test transposes
- fixes broadcasting of addmv
- not supporting tf32 yet, will do it in future PR together with other testing fixes

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -16368,53 +16385,46 @@ def _test(use_out, row_major, incx, incy, lda_tail):
...
                         torch.half: 1e-1, torch.cfloat: 1e-4, torch.cdouble: 1e-8})
     @dtypesIfCUDA(*torch.testing.get_all_complex_dtypes(), *torch.testing.get_all_fp_dtypes(include_bfloat16=False))
     @dtypes(*torch.testing.get_all_complex_dtypes(), *torch.testing.get_all_fp_dtypes())
     @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
     def test_addmm(self, device, dtype):
-        M = torch.randn(10, 25).to(device=device, dtype=dtype)
-        m1 = torch.randn(10, 50).to(device=device, dtype=dtype)
-        m2 = torch.randn(50, 25).to(device=device, dtype=dtype)
-        self._test_addmm(M, m1, m2)
+        M = torch.randn(10, 25, device=device).to(dtype)
```
Comment (ngimel) [inline test/test_torch.py:16399]: can you leave a comment here why you need `.to(dtype)`. Which system/dtype combination is faulty?

File: test/test_torch.py [hunk 2]
```diff
@@ -16368,53 +16385,46 @@ def _test(use_out, row_major, incx, incy, lda_tail):
...
+        self._test_addmm_addmv(torch.addmm, M, m1, m2)
+
+        # Test beta=0, M=nan
+        M = torch.full((10, 25), math.nan, device=device).to(dtype)
+        m1 = torch.randn(10, 50, device=device).to(dtype)
+        m2 = torch.randn(50, 25, device=device).to(dtype)
+        self._test_addmm_addmv(torch.addmm, M, m1, m2, beta=0)
+
+        # Test transpose
+        for t1, t2, t3, t4 in product([True, False], repeat=4):
```
Comment (ngimel) [inline test/test_torch.py:16417]: nice!

File: test/test_torch.py [hunk 3]
```diff
@@ -16368,53 +16385,46 @@ def _test(use_out, row_major, incx, incy, lda_tail):
...
+            M = torch.randn(10, 25, device=device).to(dtype)
+            m1 = torch.randn(10, 50, device=device).to(dtype)
+            m2 = torch.randn(50, 25, device=device).to(dtype)
+            if t1:
```
Comment (ngimel) [inline test/test_torch.py:16421]: nit: maybe factor it out into `maybe_transpose` function?

File: test/test_torch.py (hunk 1 above)
Comment (zasdfgbnm) [reply test/test_torch.py:16402]: done

File: test/test_torch.py (hunk 3 above)
Comment (zasdfgbnm) [reply test/test_torch.py]: done

File: test/test_torch.py [hunk 4]
```diff
@@ -16368,53 +16388,45 @@ def _test(use_out, row_major, incx, incy, lda_tail):
...
                         torch.half: 1e-1, torch.cfloat: 1e-4, torch.cdouble: 1e-8})
     @dtypesIfCUDA(*torch.testing.get_all_complex_dtypes(), *torch.testing.get_all_fp_dtypes(include_bfloat16=False))
     @dtypes(*torch.testing.get_all_complex_dtypes(), *torch.testing.get_all_fp_dtypes())
     @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
     def test_addmm(self, device, dtype):
-        M = torch.randn(10, 25).to(device=device, dtype=dtype)
-        m1 = torch.randn(10, 50).to(device=device, dtype=dtype)
-        m2 = torch.randn(50, 25).to(device=device, dtype=dtype)
-        self._test_addmm(M, m1, m2)
+        M = torch.randn(10, 25, device=device).to(dtype)
```
Comment (ngimel) [inline test/test_torch.py:16402]: So `(..., device=device, dtype=dtype)` does not work for bfloat16 on cpu? Interesting. `rand` works btw.

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**